ql/src/java/org/apache/hadoop/hive/ql/Compiler.java

```diff
@@ -224,6 +224,7 @@ private BaseSemanticAnalyzer analyze() throws Exception {
       sem.analyze(tree, context);
     } finally {
       sem.endAnalysis();
+      driverContext.setQueryType(sem, tree);
```

Comment:
why do we need to set QueryType in a finally?

Reply:
the service works for failed queries too, where e.g. SematicException happened, that's where the finally block kicks in, making the service create a query history record containing the exception too

Reply:
makes sense, just don't really like the way we set queryType